linux/shim/service.go

```diff
@@ -70,31 +73,27 @@ type Service struct {
 	mu            sync.Mutex
 	processes     map[string]process
 	events        chan interface{}
-	eventsMu      sync.Mutex
 	deferredEvent interface{}
 	namespace     string
 	context       context.Context
+	ec            chan runc.Exit
 
 	workDir  string
 	platform platform
 }
 
 func (s *Service) Create(ctx context.Context, r *shimapi.CreateTaskRequest) (*shimapi.CreateTaskResponse, error) {
+	s.mu.Lock()
```

Comment:
can we use `defer s.mu.Lock()` now?

Otherwise, move it back after `newInitProcess` or unlock it in the error path.

Reply:
done